user_tools/src/spark_rapids_pytools/resources/dev/prepackage_mgr.py

```diff
@@ -0,0 +1,170 @@
+        return (f'{self._mvn_base_url}/'  # pylint: disable=no-member
+                f'{jar_version}/rapids-4-spark-tools_2.12-{jar_version}.jar')
+
+    def _fetch_resources(self) -> dict:
+        """
+        Fetches the resource information from configuration files for each supported platform.
+        Returns a dictionary of resource details.
+        """
+        resource_uris = {}
+
+        # Add RAPIDS JAR as dependency
+        if self.tools_jar:
+            # copy from existing file. replace snapshot
+            jar_file_name = FSUtil.get_resource_name(self.tools_jar)
+            FSUtil.make_dirs(self.dest_dir)
+            dest_file = FSUtil.build_path(self.dest_dir, jar_file_name)
+            shutil.copy2(self.tools_jar, dest_file)
+        else:
+            # get the latest tools_jar from mvn
+            rapids_url = self._get_spark_rapids_jar_url()
+            rapids_name = FSUtil.get_resource_name(rapids_url)
+            resource_uris[rapids_url] = {'name': rapids_name, 'pbar_enabled': True}
+
+        for platform in self._supported_platforms:  # pylint: disable=no-member
+            config_file = FSUtil.build_full_path(self.resource_dir,
+                                                 f'{platform}{self._configs_suffix}')  # pylint: disable=no-member
+            platform_conf = JSONPropertiesContainer(config_file)
+            for dependency in platform_conf.get_value('dependencies', 'deployMode', 'LOCAL'):
+                uri = dependency.get('uri')
+                name = FSUtil.get_resource_name(uri)
+                if uri:
+                    resource_uris[uri] = {'name': name, 'pbar_enabled': True}
+                    resource_uris[uri + '.asc'] = {'name': name + '.asc', 'pbar_enabled': False}
+
+            # Add pricing files as resources
+            if platform_conf.get_value_silent('pricing'):
+                for pricing_entry in platform_conf.get_value('pricing', 'catalog', 'onlineResources'):
+                    uri = pricing_entry.get('onlineURL')
+                    name = pricing_entry.get('localFile')
+                    if uri and name:
+                        resource_uris[uri] = {'name': name, 'pbar_enabled': False}
+
+        return resource_uris
+
+    def _download_resources(self, resource_uris: dict):
+        resource_uris_list = list(resource_uris.items())
+
+        def download_task(resource_uri, resource_info):
+            resource_name = resource_info['name']
+            pbar_enabled = resource_info['pbar_enabled']
+            resource_file_path = FSUtil.build_full_path(self.dest_dir, resource_name)
+
+            print(f'Downloading {resource_name}')
+            FSUtil.fast_download_url(resource_uri, resource_file_path, pbar_enabled=pbar_enabled)
+
+        with ThreadPoolExecutor() as executor:
+            executor.map(lambda x: download_task(x[0], x[1]), resource_uris_list)
+
+    def _compress_resources(self) -> Optional[str]:
```

Comment:
nit: We are using `Optional[str]` for type here. Are we expecting a `None` type return in this method?